ragstack-e2e-tests/e2e_tests/llama_index/test_compatibility_rag.py

```diff
@@ -27,37 +34,120 @@
     get_astra_ref,
     delete_all_astra_collections,
     delete_astra_collection,
+    AstraRef,
 )
+from llama_index.vector_stores.types import VectorStore, VectorStoreQuery
+from sqlalchemy import Float
+from vertexai.vision_models import MultiModalEmbeddingModel, Image
+
+
+class VectorStoreWrapper:
+    @abstractmethod
+    def init(self, embedding_dimension: int) -> str:
+        pass
+
+    @abstractmethod
+    def as_vector_store(self) -> VectorStore:
```

Comment:
We could do an abstract class rather than a pure interface.

Reply:
I did it on purpose to being able to catch errors in the actual init method and not in the fixture construction